misk/src/test/kotlin/misk/web/actions/AdminDashboardTabActionTest.kt

```diff
@@ -0,0 +1,74 @@
+package misk.web.actions
+
+import misk.client.HttpClientEndpointConfig
+import misk.client.HttpClientFactory
+import misk.security.authz.FakeCallerAuthenticator
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import misk.web.jetty.JettyService
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import javax.inject.Inject
+
+@MiskTest(startService = true)
+class AdminDashboardTabActionTest {
+  @MiskTestModule
+  val module = AdminDashboardActionTestingModule()
+
+  @Inject private lateinit var jetty: JettyService
+  @Inject private lateinit var httpClientFactory: HttpClientFactory
+  @Inject private lateinit var adminDashboardTabAction: AdminDashboardTabAction
+
+  val path = "/api/admindashboardtabs"
+  val empty = "{\"adminDashboardTabs\":[]}"
+  val tabs =
+    "{\"adminDashboardTabs\":[{\"slug\":\"config\",\"url_path_prefix\":\"/_admin/config/\",\"name\":\"Config\",\"category\":\"Container Admin\",\"capabilities\":[\"admin_access\"],\"services\":[]},{\"slug\":\"web-actions\",\"url_path_prefix\":\"/_admin/web-actions/\",\"name\":\"Web Actions\",\"category\":\"Container Admin\",\"capabilities\":[\"admin_access\"],\"services\":[]}]}"
```

Comment:
pretty print this JSON and use a multiline string?

Reply:
and then you can assert on the object, not on its string representation?
```
val moshi = Moshi.Builder().build()
val responseAdapter = moshi.adapter<AdminDashboardTabAction.Response>()
return responseAdapter.fromJson(response.body!!.source())
```

Reply:
Perfect! I wanted to do this but I didn't know how 👍